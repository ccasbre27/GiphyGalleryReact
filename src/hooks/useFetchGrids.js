import { useState, useEffect } from "react"
import { getGifs } from '../helpers/getGifs';

export const useFetchGrids = (category) => {

    const [state, setState] = useState({
        data: [],
        loading: true
    })    

    useEffect( () => {
        getGifs( category )
            .then( imgs => {

                setTimeout( () => {
                    
                    setState({ data: imgs, loading: false }) ;
                }, 3000);
            })
                
    }, [ category ]); // el arreglo vacío indica que sólo se renderiza una vez


    return state;
}