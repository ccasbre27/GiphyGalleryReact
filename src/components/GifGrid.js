import React from 'react'
import { GifGridItem } from './GifGridItem';
import { useFetchGifs } from '../hooks/useFetchGifs';
import PropTypes from 'prop-types';

export const GifGrid = ({ category }) => {

    const { data: images, loading } = useFetchGifs( category );

    return (
        <div>
            <h3> { category } </h3>
            { loading && <p> Loading </p> }
            <div className="card-grid">
                <ol>
                    {
                        images.map( img => {
                            return <GifGridItem 
                                key={ img.id }
                                { ...img } />
                        })
                    }
                </ol>
            </div>
       </div>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
};
