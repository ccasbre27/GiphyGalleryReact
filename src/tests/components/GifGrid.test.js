import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import React from 'react';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs');

describe('Tests on <GifGrid />', () => {
   
    const category = 'Dragon Ball';
    const gifs = [
        {
            id: 'ABC',
            url: 'https://localhost/something',
            title: 'Lorem ipsum'
        },
        {
            id: 'ABCD',
            url: 'https://localhost/something',
            title: 'Lorem ipsum'
        }
    ];

    beforeEach(() => {
        useFetchGifs.mockImplementation( category => {
            return { data: gifs, loading: false }
          });
    });
        
    test('should render the component', () => {
       
        const wrapper = shallow( <GifGrid category={ category } /> );

        expect( wrapper ).toMatchSnapshot();
        
    });
    
    test('should render items when the images are returned', () => {
       
        const wrapper = shallow( <GifGrid category={ category } /> );

        expect( wrapper.find('p').exists() ).toBe(false);
        expect( wrapper.find('GifGridItem').length ).toBe(gifs.length);

    });
    

});
