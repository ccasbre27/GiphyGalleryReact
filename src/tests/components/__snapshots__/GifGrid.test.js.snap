// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <GifGrid /> should render the component 1`] = `
<div>
  <h3>
     
    Dragon Ball
     
  </h3>
  <div
    className="card-grid"
  >
    <ol>
      <GifGridItem
        id="ABC"
        key="ABC"
        title="Lorem ipsum"
        url="https://localhost/something"
      />
      <GifGridItem
        id="ABCD"
        key="ABCD"
        title="Lorem ipsum"
        url="https://localhost/something"
      />
    </ol>
  </div>
</div>
`;
